notebooks/03.02. Supervised Learning.Rmd

```diff
@@ -309,24 +312,312 @@ filtered_onehot_test <- read.csv("../data/processed/test/filtered_onehot_test.cs
 
 ### Random Forest
 
+This method is a combination of tree predictors such that each tree depends on the values of a random vector sampled independently and with the same distribution for all trees in the forest. Ranger is a fast implementation of Random Forest for high-dimensional data.
+
+Tuning parameters:
+- `mtry` (#Randomly Selected Predictors)
+- `splitrule` (Splitting Rule)
+- `min.node.size` (Minimal Node Size)
+
+```{r}
+# Define the tuning grid
+rf_filtered_tuneGrid <- expand.grid(.mtry = c(2, 3, 4, 5, 6, 7, 8, 9, 10),
+                        .splitrule = "gini",
+                        .min.node.size = c(1, 5, 10, 15, 20))
+
+# Define the model and predictions paths
+rf_filtered_model_path <- "../models/rf_filtered_onehot_model.RData"
+rf_filtered_predictions_path <- "../data/predictions/rf_filtered_onehot_predictions.csv"
+
+# Apply the algorithm
+rf_filtered_model <- apply_algorithm("ranger", rf_filtered_tuneGrid, filtered_onehot_train, filtered_onehot_test, rf_filtered_model_path, rf_filtered_predictions_path)
+```
+
+Now we can evaluate the model using the confusion matrix and other metrics.
+
+```{r}
+# Load the predictions
+rf_filtered_onehot_predictions <- read.csv(rf_filtered_predictions_path)
+
+# Get the metrics
+metrics_rf_filtered <- get_prediction_metrics(filtered_onehot_test, rf_filtered_onehot_predictions)
+metrics_rf_filtered$confusion
+metrics_rf_filtered$metrics_table
+```
+
+Finally, we can calculate the AUC.
+
+```{r}
+# Get the AUC
+auc_rf_filtered <- get_AUC_multiclass(filtered_onehot_test, rf_filtered_onehot_predictions)
```

Comment:
In order to keep consistency, I think we should rename the metrics and AUC variables so that the algorithm goes first, then the dataset and finally the corresponding name (metrics or auc). 

For example: `metrics_rf_filtered` should be `rf_filtered_metrics`